examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,231 @@
+            argv[0]);
+        return 1;
+    }
+
+    char *remote_ip = argv[1];
+    int port = atoi(argv[2]);
+
+    /* ensure client isn't connected in case of a new connection */
+    if (client.isconnected) {
+       printf("client already connected, disconnecting it\n");
+       MQTTDisconnect(&client);
+    }
+
+    if (!is_initialized) {
+        MQTTNetworkInit(&network);
+        printf("Launching MQTT Task\n");
+        MQTTStartTask(&client);
+        is_initialized = true;
+    }
+
+    printf("Trying to connect to %s , port: %d\n",
+           remote_ip, port);
+    if (MQTTNetworkConnect(&network, remote_ip, port)) {
+        printf("error: Unable to connect to %s:%d\n", remote_ip,
+               port);
+        return 1;
+    }
+
+    MQTTClientInit(&client, &network, COMMAND_TIMEOUT_MS, buf, BUF_SIZE, readbuf,
+                   BUF_SIZE);
+
+    MQTTPacket_connectData data = MQTTPacket_connectData_initializer;
+    data.MQTTVersion = MQTT_VERSION;
+    data.clientID.cstring = argv[3];
+    data.username.cstring = argv[4];
+    data.password.cstring = argv[5];
+    data.keepAliveInterval = atoi(argv[6]);
+    data.cleansession = IS_CLEAN_SESSION;
+    printf("Connecting to %s %d\n", remote_ip, port);
+
+    int ret = MQTTConnect(&client, &data);
+    if (ret) {
+        printf("error: Unable to connect client %d\n", ret);
+    }
+    else {
+        printf("Connection successfully\n");
+    }
+    return ret;
+}
+
+static int _cmd_discon(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    int res = MQTTDisconnect(&client);
+    if (res) {
+        puts("error: Unable to disconnect");
+    }
+    else {
+        puts("Disconnect successful");
+    }
+
+    MQTTNetworkDisconnect(&network);
+    return res;
+}
+
+static int _cmd_pub(int argc, char **argv)
+{
+    enum QoS qos = QOS0;
+
+    if (argc < 3) {
+        printf("usage: %s <topic name> <string msg> [QoS level]\n",
+               argv[0]);
+        return 1;
+    }
+    if (argc == 4) {
+        qos = get_qos(argv[3]);
+    }
+    MQTTMessage message;
+    message.qos = qos;
+    message.retained = NOT_RETAINED_MSG;
+    message.payload = argv[2];
+    message.payloadlen = strlen(message.payload);
+
+    int rc;
+    if ((rc = MQTTPublish(&client, argv[1], &message)) != 0) {
+        printf("error: Unable to publish (%d)\n", rc);
+    }
+    else {
+        printf("Message (%s) has been published to topic %s\n",
+               (char *)message.payload, argv[1]);
+    }
+
+    return rc;
+}
+
+static int _cmd_sub(int argc, char **argv)
+{
+    enum QoS qos = QOS0;
+
+    if (argc < 2) {
+        printf("usage: %s <topic name> [QoS level]\n", argv[0]);
+        return 1;
+    }
+
+    if (argc >= 3) {
+        qos = get_qos(argv[2]);
+    }
+
+    printf("Subscribing to %s\n", argv[1]);
+    int ret = MQTTSubscribe(&client, argv[1], qos, _on_msg_received);
+    printf("Subscribed %d\n", ret);
+
+    if (!ret) {
```

Comment:
```suggestion
    if (ret < 0) {
```

According to the return code of the paho mqtt functions:
https://github.com/eclipse/paho.mqtt.embedded-c/blob/29ab2aa29c5e47794284376d7f8386cfd54c3eed/MQTTClient/src/MQTTClient.h#L46